tutorials/Land/Soil/Coupled/equilibrium_test.jl

```diff
@@ -424,101 +391,74 @@ boundaries = [
     FT(1) FT(1) zmax
 ]
 resolution = (FT(2), FT(2), zres)
-thegrid = solver_config.dg.grid
-intrp_brck = create_interpolation_grid(boundaries, resolution, thegrid)
-step = [1];
-callback = GenericCallbacks.EveryXSimulationTime(
-    every_x_simulation_time,
-) do (init = false)
-    t = ODESolvers.gettime(solver_config.solver)
-    iQ, iaux, igrads = interpolate_variables((Q, aux, grads), intrp_brck)
-    ϑ_l = iQ[:, ϑ_l_ind, :][:]
-    T = iaux[:, T_ind, :][:]
-    K∇h_vert = igrads[:, K∇h_vert_ind, :][:]
-    κ∇T_vert = igrads[:, κ∇T_vert_ind, :][:]
-    all_vars = Dict{String, Array}(
-        "t" => [t],
-        "ϑ_l" => ϑ_l,
-        "T" => T,
-        "K∇h_vert" => K∇h_vert,
-        "κ∇T_vert" => κ∇T_vert,
-    )
-    all_data[step[1]] = all_vars
 
-    step[1] += 1
+callback = GenericCallbacks.EveryXSimulationTime(every_x_simulation_time) do
+    dons = dict_of_nodal_states(solver_config, state_types; interp = true)
```

Comment:
question re: how this does interpolations...
before we were doing something to conform with this discussion (what is used in Diagnostics)
https://github.com/CliMA/ClimateMachine.jl/issues/1092

I assume this is creating an interpolation function and then evaluating that at the unique z values, but wondering if that interpolation function is different from what we were using before. I think it boiled down to a simple mean in the previous case, but that from the conversation on that issue I gathered it could be more complex for other geometries.

Reply:
This only interpolates the face elements, it does not interpolate onto a regular grid.